mkosi/__init__.py

```diff
@@ -1334,69 +1347,75 @@
 
 
 def print_summary(config: MkosiConfig) -> None:
-    print("COMMANDS:")
+    f = io.StringIO()
+    b = MkosiPrinter.bold
+    e = MkosiPrinter.reset
+    say: Callable[..., None] = lambda *args, **kwargs: print(*args, **kwargs, file=f)
```

Comment:
Awesome!
I hope to understand this soon.
But I love that it's Python.

Reply:
That warning's a bit spurious, more of a style thing, since `say("foo")` is equivalent to `print("foo", file=f)` it argues the latter should be used. I think one can have different opinions on that topic. For the most part we ignore the CodeQL stuff, since the vast majority of hits are for preexisting code that we don't want to touch at that moment.

Reply:
I read the documentation that is linked, and it explicitly wants to cover the case where arguments are passed through unchanged. But in our case, they *are* changed. I don't think codeql is trying to be smart and intentionally suggest a refactoring that moves the `file` argument. I think the warning is just badly implemented, and somebody messed up the check for "arguments are passed through unchanged".